nnpy/errors.py

```diff
@@ -1,12 +1,16 @@
 from _nnpy import ffi, lib as nanomsg
 
 class NNError(Exception):
-    pass
+    def __init__(self, result_code, error_no, *args, **kwargs):
```

Comment:
Not sure it makes sense to pass in `result_code` here. Do you need the `*args, **kwargs` for `aionn`? Otherwise I'd rather leave them out for now.